bindings/rust/aws-lc-sys-template/build.rs

```diff
@@ -0,0 +1,318 @@
+    builder
+}
+
+const AWS_LC_PATH: &str = "deps/aws-lc";
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+enum OutputLib {
+    Crypto,
+    Ssl,
+}
+
+#[allow(dead_code)]
+#[derive(Clone, Copy, PartialEq, Eq)]
+enum OutputLibType {
+    Static,
+    Dynamic,
+}
+
+impl OutputLibType {
+    fn rust_lib_type(&self) -> &str {
+        match self {
+            OutputLibType::Static => "static",
+            OutputLibType::Dynamic => "dylib",
+        }
+    }
+}
+
+impl OutputLib {
+    fn libname(&self, prefix: Option<&str>) -> String {
+        format!(
+            "{}{}",
+            if let Some(pfix) = prefix.to_owned() {
+                pfix
+            } else {
+                ""
+            },
+            match self {
+                OutputLib::Crypto => "crypto",
+                OutputLib::Ssl => "ssl",
+            }
+        )
+    }
+
+    fn locate_dir(&self, path: &Path) -> PathBuf {
+        path.join(Path::new(&format!("build/{}", self.libname(None))))
+            .join(get_platform_output_path())
+    }
+}
+
+fn get_platform_output_path() -> PathBuf {
+    PathBuf::new()
+}
+
+fn get_cmake_config() -> cmake::Config {
+    let pwd = env::current_dir().unwrap();
+
+    cmake::Config::new(pwd.join(AWS_LC_PATH))
+}
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+fn prefix_string() -> String {
+    format!("aws_lc_{}", VERSION.to_string().replace('.', "_"))
+}
+
+fn verify_fips_clang_version() -> (&'static str, &'static str) {
+    fn version(tool: &str) -> String {
+        let output = match Command::new(tool).arg("--version").output() {
+            Ok(o) => o,
+            Err(e) => {
+                eprintln!("warning: missing {}, trying other compilers: {}", tool, e);
+                // NOTE: hard-codes that the loop below checks the version
+                return String::new();
+            }
+        };
+        assert!(output.status.success());
+        let output = std::str::from_utf8(&output.stdout).expect("invalid utf8 output");
+        output.lines().next().expect("empty output").to_string()
+    }
+
+    const REQUIRED_CLANG_VERSION: &str = "7.0.1";
+    for (cc, cxx) in [
+        ("clang-7", "clang++-7"),
+        ("clang", "clang++"),
+        ("cc", "c++"),
+    ] {
+        let cc_version = version(cc);
+        if cc_version.contains(REQUIRED_CLANG_VERSION) {
+            assert!(
+                version(cxx).contains(REQUIRED_CLANG_VERSION),
+                "mismatched versions of cc and c++"
+            );
+            return (cc, cxx);
+        } else if cc == "cc" {
+            panic!(
+                "unsupported clang version \"{}\": FIPS requires clang {}",
+                cc_version, REQUIRED_CLANG_VERSION
+            );
+        } else if !cc_version.is_empty() {
+            eprintln!(
+                "warning: FIPS requires clang version {}, skipping incompatible version \"{}\"",
+                REQUIRED_CLANG_VERSION, cc_version
+            );
+        }
+    }
+    unreachable!()
```

Comment:
Might be better to provide an error message here indicating that it couldn't find a c compiler?

Reply:
Good point. Will fix...

Reply:
I removed the function entirely.  It relates to a FIPS build, which we don't yet support anyways.